fiftyone/core/view.py

```
@@ -920,6 +920,19 @@ def get_dynamic_group(self, group_value):
         if etau.is_str(group_expr):
             pipeline.append({"$match": {group_expr[1:]: group_value}})
+        elif isinstance(group_expr, (list, tuple)) and all(
+            etau.is_str(e) and e.startswith("$") for e in group_expr
+        ):
+            # match each field directly (index-eligible); an `$expr` over a
+            # computed array cannot use an index
+            pipeline.append(
+                {
+                    "$match": {
+                        field_ref[1:]: value
+                        for field_ref, value in zip(group_expr, group_value)
```

Ruff (0.15.18)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)